src/model/nets/models_genesis.py

```diff
@@ -56,13 +58,36 @@ def __init__(self, in_channels, out_channels, weight_path=None, loaded_modules=N
                 all_modules = {'in_block', 'down_block1', 'down_block2', 'down_block3',
                                'up_block1', 'up_block2', 'up_block3'}
                 assert loaded_modules.issubset(all_modules)
-
+                
                 for key in list(pretrained_state_dict.keys()):
                     if not any(loaded_module in key for loaded_module in loaded_modules):
                         pretrained_state_dict.pop(key)
+
+            if weight_settings is not None:
+                percentage = weight_settings.percentage
```

Comment:
Any reason to set `weight_settings` to `BoxList`? Maybe a float arg (e.g., `weight_percentage`) is enough.

Reply:
For the extendibility. There may be other settings related to the weights in the future.